addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java

```diff
@@ -257,6 +257,10 @@ public void scan() {
                 }
                 sendAndReceive(hackedMessage, true); // follow redirects
 
+                if (isPage404(hackedMessage)) {
```

Comment:
Shouldn't we ignore non-HTML content-types instead of the status code?

Reply:
status code is definitely needed I believe. But I think we can add a check for non-HTML content-types too.

Reply:
https://hypejabbed.getastra.dev/WebInterface/login.html%23/WebInterface/login.html%23 this URL raised a false positive for us. Although I think you won't be able to reproduce this.

Reply:
Is that HTML originally? I'm seeing a 404 and JSON content-type.

Reply:
Yes I think it is no longer accessible. This was the evidence which we got which is HTML content. 

`<form action="loginVerify" method="POST" class="form-signin"> <input type="text" name="username" id="inputUsername" class="form-control form-group" placeholder="email" required autofocus> <input type="password" name="password" id="inputPassword" class="form-control form-group" placeholder="password" required> <button class="btn btn-lg btn-primary btn-block btn-signin" type="submit">enter</button> </form>`

Reply:
But was the content-type HTML? Do you still have the session?

Reply:
let me check

Reply:
The other info should have more details.

Reply:
But thinking it logically I think it is still possible to reproduce such type of scenario

Reply:
I think you are right we could just check for non HTML content too.

Reply:
I think we can add a check to check HTML. I don't have an example to prove but I think this can easily be produced where a HTML and 404 causes a FP

Reply:
After thinking a lot about this, I think you are right. We can just simply ignore non-html pages.